ray-operator/apis/ray/v1alpha1/raycluster_types.go

```diff
@@ -33,6 +33,11 @@ type HeadGroupSpec struct {
 	// Number of desired pods in this pod group. This is a pointer to distinguish between explicit
 	// zero and not specified. Defaults to 1.
 	Replicas *int32 `json:"replicas"`
+	// RayCustomResources is a string-int map representing optional, user-specified custom resource
+	// capacities of Ray pods in this group.
+	// Equivalent to RayStartParams.resources, which must be specified as a json string.
+	// Disallowed keys: "GPU", "CPU", "memory", "object_store_memory". Instead, use rayStartParams["num-cpus"] etc.
+	RayCustomResources map[string]int64 `json:"rayCustomResources,omitempty"`
```

Comment:
I am thinking in the real world cases, normally we need to claim the resource twice ( kubernetes level and ray level). for example, If I need an inference chip like `aws.amazon.com/neuron: 1`.  

kubernetes level configuration
```
resources:
    limits:
       aws.amazon.com/neuron: 1
```

ray level configuration
```
rayCustomResources
     aws.amazon.com/neuron: 1
```

Is that expected? Not sure if the resource name are always same between kubernetes and ray. If so, we probably can use kubernetes resource as source of truth to build ray resource configurations. If not, I think separation is necessary.  (I kind of forget our conclusion in last community meeting)

Reply:
For now, I'd prefer to keep the configuration separate -- this PR isn't introducing new behavior, just cleaning up the existing interface.
`rayCustomResources` is an alias for rayStartParams["resources"]. You use a map instead of a triply-quoted json string.

For the use-case described in https://github.com/ray-project/kuberay/issues/167 (multiple image types) it's not needed to set up a Kubernetes extended resource. 
The Ray resource annotation is just used as a hint to the Ray scheduler to schedule certain tasks on workers using certain images. 
For use-cases like this, it's too heavy-weight to require the user to patch nodes with extended resource capacities (or run a daemonset to do the same).

In the other direction, **if your Ray pod already uses K8s extended resources**, it could be useful for Ray to automatically detect those as "custom resources". Right now, we do that only with "/gpu" extended resources, but we might want to generalize the behavior. We could make this generalization later, if that's desired by users.

Reply:
Summarizing my opinion:

- If the pod uses K8s extended resource, it may be useful to automatically advertise it to Ray as a "custom resource". (But we're not doing that in this PR.)

- Advertising a "custom resource" to Ray should not require labelling K8s nodes with extended resources.

Reply:
Summarize looks good to me.